src/broker/overlay.c

```diff
@@ -1249,13 +1250,44 @@ static void parent_monitor_cb (struct zmqutil_monitor *mon, void *arg)
     }
 }
 
+/* Zeromq treats failed hostname resolution as a transient error, and silently
+ * retries in the background, which can make config problems hard to diagnose.
+ * Parse the URI in advance, and if the host portion is invalid, log it.
+ * Ref: flux-framework/flux-core#5009
+ */
+static void warn_of_invalid_host (flux_t *h, const char *uri)
+{
+    char *cpy;
+    struct yuarel u = { 0 };
+    struct addrinfo *result;
+    int e;
+
+    if (!(cpy = strdup (uri))
+        || yuarel_parse (&u, cpy) < 0
+        || !u.scheme
+        || !u.host
+        || !streq (u.scheme, "tcp"))
+        goto done;
+    /* N.B. this URI will be used for zmq_connect(), therefore it must
+     * be a valid peer address, not an interface name or wildcard.
+     */
+    if ((e = getaddrinfo (u.host, NULL, NULL, &result)) == 0) {
+        freeaddrinfo (result);
+        goto done;
+    }
+    flux_log (h, LOG_ERR, "getaddrinfo %s: %s", u.host, gai_strerror (e));
```

Comment:
Just one quick comment before a full review: Users and admins may appreciate a more descriptive message here like: "Warning: unable to resolve upstream peer %s: %s"?

Reply:
Yeah probably :-)  Will fix.

Reply:
Force pushed a fix to that one.  Good call!